docs/custom-directive.md

```diff
@@ -0,0 +1,85 @@
+# Custom directive
+A GraphQL directive is a special syntax used to provide additional information to the GraphQL execution engine about how to process a query, mutation, or schema definition. For example, directives can be used to modify the behaviour of fields, arguments, or types in your schema. 
+
+A custom directive is composed of 2 parts:
+- schema definitions
+- schema transformer
+
+## Schema Definition
+It is the syntax used to describe a custom directive within a GraphQL schema.
+To define a custom directive, you must use the directive keyword, followed by its name, arguments (if any), and the locations where it can be applied. 
```

Comment:
I'd link to some docs explaining what possible locations there are besides FIELD_DEFINITION